### ISSUE TITLE
Scaling bug with mastery vs spell power, resulting in low damage for casters that prioritize mastery 

### ISSUE THREAD

**Issue by Hatem902** [gamefreedomgit/Maelstrom#4032]
There seems to be a bug which nerfs casters that use mastery, likely because of scaling issues. 

**Experiment :** 
To check out trinkets for my frost mage, I tested a few, mainly mirrors and the bell trinket. I equipped both at the same time: 359 bell and 359 mirrors. But only left one proc at a time in isolation.

The test was pretty straightforward: just one ice lance with FoF on a dummy.

In the past, back in Apollo, I did this to see which proc would give me more damage. Turns out, **1926 mastery** (Frostburn - Increased damage Vs frozen targets by a certain percentage.) **gave a lot more damage than 1926 spell power** with season 9 gear. I did the same thing also in WKMZ's server (but 372), and I got the same result.

I thought I didn't need to test this again until I noticed my frost mage is dealing way less damage than normal.

So, I decided to do the exact same thing again here on my frost mage in whitemane. And these were the results:

1926 mastery proc --> average of 22k ice lance crit
1926 spell power proc --> average of 24.5k ice lance crit

**TLDR:** 
Because of end-game scaling being applied on season 9 gear, casters that use Mastery Trinkets or that rely on mastery as a main stat for damage, got nerfed. (No wonder that not a single caster spec that prioritizes mastery, is currently being played in high rated arena)

And the thing with mastery is that some caster specs rely completely on it for damage ( one of the reasons frost mage was a strong spec before was because of this trinket [Foul Gift of the Demon Lord](https://www.wowhead.com/item=72898/foul-gift-of-the-demon-lord) which comes in future seasons, (and/or Theralion's Mirrors)).

For now, when Theralion's mirror procs, Frost mage is dealing about 20% reduced damage compared to what it's supposed to. That added to the fact that mastery is a main stat that you equip in other gear pieces, makes Frost mage damage noticeably very low because of this scaling bug with mastery vs spell power.

Could you guys please look into this bug and fix it asap, **It's very easy to reproduce in the test server and contemplate the scaling issue with mastery vs spell power for casters.**

**Comment by Pornoodemon** [comment]
I checked today on the test server. Mastery correctly increases damage.
And spell power as a percentage has always given more damage than mastery, since spell power multiplies with buffs, talents, passive skills and other things.

**Comment by Hatem902** [comment]
> I checked today frost mage on the test server. Mastery correctly increases damage. And spell power as a percentage has always given more damage than mastery, since spell power multiplies with buffs, talents, passive skills and other things.
> 
> Another example in another talent tree (Demonology) Proc for 3200 Mastery (Increases damage by 60%), however, the trinket for 2500 spell power gives exactly the same numbers in damage

We're on the same page with the fact that Spell power is currently giving more damage than Mastery with a normal s9 gear.
But is it supposed to? as I told you I tested on other servers ( just for the purpose of choosing trinkets ) and Mastery has always given more damage than spell power point per point for frost mage.

Mastery is supposed to take the output damage after counting all buffs, talents, passive skills, then it multiplies the whole thing by a percentage (let's say 20%). Example : 35k ----->(buffs,talents, spellpower increases etc.) =  40k ----->(20% dmg increase from mastery) = 48k.
Also by nature it's supposed to do more damage point for point because it only applies on certain conditions (for example only on frozen target as opposed to spell power which applies without any condition).

**Comment by Pornoodemon** [comment]
![image](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/36e95261-7745-4791-91dd-d6ecabee3b46)

7 Mastery (Damage increased by 16%)
Everything is working fine
![image](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/2c89c349-9194-416c-9f5c-e552ea7a7eee)

![image](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/c134ba71-c4f2-4d89-a2a7-d197519e5e7f)
![image](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/8cf235c4-3c3e-46ab-8879-418ff11bf012)

**Comment by Pornoodemon** [comment]
I mean there is no problem with Mastery.
If it seems to you that something is wrong, then look for the problem in something else.

**Comment by Hatem902** [comment]
> I mean there is no problem with Mastery. If it seems to you that something is wrong, then look for the problem in something else.

Yeah I'm with you, that's what i said initially "There seems to be a bug which nerfs casters that use mastery, **likely because of scaling issues.**"

**Comment by UltraNix** [comment]
Also tested. Mastery gives damage what it should.
Spell damage depends on many factors, including spell coefficients. And thus depens on how much SP do you have.
Spell power != mastery. 
E.g. Ice Lance
1926 SP gives you +~700dmg (+25%) on "clean" char.
1926 mastery gives you +~800dmg (+28%) on "clean" char.
Both calculations are okay. Nothing to say more about it.
